.github/scripts/validate_scale_config.py

```diff
@@ -98,47 +137,83 @@ def runner_types_are_equivalent(
-            amz2023_runner_type = f"{AMAZON_2023_PREFIX}{runner_type}"
+        # Ensure that the max_available is at least 50
+        # this is a requirement as scale-up always keeps at minimum some spare runners live, and less than 50
+        # will very easily trigger alerts of not enough runners
+        if runner_config["max_available"] < 50:
+            print(
+                f"Runner type {runner_type} has max_available set to {runner_config['max_available']}, "
+                f"which is less than the minimum required value of 50"
+            )
+            errors_found = True
 
-            if amz2023_runner_type not in runner_types:
-                print(
-                    f"Runner type {base_runner_type} does not have a corresponding {amz2023_runner_type} runner type"
-                )
-                errors_found = True
-                continue
+        # Next validations are for linux runners only
+        if runner_config.get("os").lower() == "windows":
+            continue
 
-        else:
-            base_runner_type = runner_type[len(AMAZON_2023_PREFIX) :]
-            amz2023_runner_type = runner_type
+        # Validations now are to guarantee that Amazon 2023 is a variant of the runner type
+        # this is why we `continue` on errors instead of only setting errors_found to True
+        # this should be the last validation for this runner type
+        if "variants" not in runner_config:
+            print(f"Runner type {runner_type} does not contain any variants")
+            errors_found = True
+            continue
 
-            if base_runner_type not in runner_types:
-                print(
-                    f"Runner type {amz2023_runner_type} does not have a corresponding {base_runner_type} runner type"
-                )
-                errors_found = True
-                continue
-
-        if not runner_types_are_equivalent(
-            base_runner_type,
-            runner_types[base_runner_type],
-            amz2023_runner_type,
-            runner_types[amz2023_runner_type],
-            ignore_ami=True,
-        ):
+        if AMAZON_2023_PREFIX not in runner_config["variants"]:
+            print(
+                f"Runner type {runner_type} does not have a corresponding {AMAZON_2023_PREFIX} variant"
+            )
+            errors_found = True
+            continue
+
+        amzn_variant = runner_config["variants"][AMAZON_2023_PREFIX]
+
+        if "ami" not in amzn_variant:
+            print(
+                f"Runner type {runner_type} variant {AMAZON_2023_PREFIX} does not have an ami"
+            )
+            errors_found = True
+            continue
+
+        if "ami" in runner_config and amzn_variant["ami"] == runner_config["ami"]:
+            print(
+                f"Runner type {runner_type} variant {AMAZON_2023_PREFIX} has the same ami as the runner type"
+            )
+            errors_found = True
+            continue
```

Comment:
This should prob be removed given our plan is now to have both the default runner type and it's amzn2023 variant serve the same AMI, right?

Reply:
Sure, but lets do this in a 2nd step, in the meantime we can guarantee the code quality. I don't believe that those 6 lines would be a huge blocker to include in that change